hooks/configure

```diff
@@ -14,119 +14,160 @@
-     ok=1
-}
-{
-    print
-}
-END {
-    if (!ok)
-        printf("%s=%s\n",ENVIRON["key"],ENVIRON["value"])
-}
-EOF
-            mv "${path}.tmp" "$path"
-        fi
-    fi
-}
-
-for service in $SERVICES; do
-    value=$(snapctl get service.$service.disable)
-    if [ -n "$value" ]; then
-        switch_service "$value" "$service"
-    fi
-done
-
-powerkey=$(snapctl get system.power-key-action)
-if [ -n "$powerkey" ]; then
-    switch_handle_power_key "$powerkey"
-fi
-
-PI_CONFIG_KEYS="disable_overscan framebuffer_width framebuffer_height
- framebuffer_depth framebuffer_ignore_alpha overscan_left overscan_right
- overscan_top overscan_bottom overscan_scale display_rotate hdmi_group
- hdmi_mode hdmi_drive avoid_warnings gpu_mem_256 gpu_mem_512 gpu_mem"
-PI_CONFIG="${TEST_UBOOT_CONFIG:-/boot/uboot/config.txt}"
-if [ -f "$PI_CONFIG" ]; then
-    for key in $PI_CONFIG_KEYS; do
-        value=$(snapctl get "pi-config.$(echo "$key"|tr _ -)")
-        update_pi_config_value "$PI_CONFIG" "$key" "$value"
-    done
-fi
+SERVICES=("ssh",)
+
+@contextmanager
+def atomic(path):
+    """atomic is a context manager that creates a new tmpfile for the
+    given target and syncs/renames it when all writing is done, to
+    make the final operation as atomic as possible.
+    """
+    dirname, target = os.path.split(path)
+    tmp = target + ".tmp"
+    dfd = os.open(dirname, os.O_RDONLY)
+    fd = os.open(tmp, os.O_WRONLY|os.O_CREAT|os.O_EXCL, mode=0o644, dir_fd=dfd)
+
+    f = open(fd, "w", closefd=False)
+
+    try:
+        yield f
+    except:
+        f.close()
+        os.close(fd)
+        os.unlink(tmp, dir_fd=dfd)
+        raise
+    else:
+        f.flush()
+        f.close()
+        os.fsync(fd)
+        os.close(fd)
+        os.rename(tmp, target, src_dir_fd=dfd, dst_dir_fd=dfd)
+    finally:
+        try:
+            os.fsync(dfd)
+        finally:
+            os.close(dfd)
+
+def systemctl(*args, check=False):
+    res = subprocess.run(["systemctl"]+list(args), stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    if check:
+        res.check_returncode()
+    return res
+
+
+def core_support_available():
+    """Return true if core_support interface is available
+
+    The check is done indirectly via checking if systemctl can be used.
+    """
+    return systemctl("--version").returncode == 0
+
+
+def switch_handle_power_key(action):
+    valid_actions = ("ignore","poweroff","reboot","halt","kexec","suspend","hibernate","hybrid-sleep","lock")
+    login_conf_d=os.getenv("TEST_LOGIN_CONF_D", default="/etc/systemd/logind.conf.d")
+    conf=os.path.join(login_conf_d, "00-snap-core.conf")
+
+    if not os.path.isdir(login_conf_d):
+        os.makedirs(login_conf_d)
+    if not action in valid_actions:
+        raise Exception("invalid action {!r} supplied for system.power-key-action option".format(action))
+    with atomic(conf) as fp:
+        fp.write("[Login]\nHandlePowerKey={}\n".format(action))
```

Comment:
You can also just `print("[Login]\nHandlePowerKey={}".format(action), file=fp)` but not much difference